defelement/implementations/fiat.py

```diff
@@ -146,63 +140,40 @@ def verify(
-            for i in range(3):
-                edofs[1][i] = edofs[1][i][:3]
-            edofs[2][0] = []
-            return edofs, lambda points: list(e.tabulate(0, points).values())[0].T.reshape(
-                points.shape[0], value_size, -1)[:, :, :9]
+            reduced_dim = 9
+
         if element.name == "reduced Hsieh-Clough-Tocher" and example.startswith("triangle"):
-            for i in range(3):
-                edofs[1][i] = []
-            return edofs, lambda points: list(e.tabulate(0, points).values())[0].T.reshape(
-                points.shape[0], value_size, -1)[:, :, :9]
+            reduced_dim = 9
+
+        if element.name == "Arnold-Winther" and example.startswith("triangle"):
+            reduced_dim = 24
+
         if element.name == "nonconforming Arnold-Winther" and example.startswith("triangle"):
-            for i in range(3):
-                edofs[1][i] = edofs[1][i][:4]
-            return edofs, lambda points: list(e.tabulate(0, points).values())[0].T.reshape(
-                points.shape[0], value_size, -1)[:, :, :15]
-        if element.name == "Guzman-Neilan (first kind)":
-            raise NotImplementedError()
-
-            if example == "triangle,1":
-                for i in range(3):
-                    edofs[1][i] = edofs[1][i][:1]
-                return edofs, lambda points: list(e.tabulate(0, points).values())[0].T.reshape(
-                    points.shape[0], value_size, -1)[:, :, :9]
-            if example == "tetrahedron,1":
-                for i in range(4):
-                    edofs[2][i] = edofs[2][i][:1]
-                return edofs, lambda points: list(e.tabulate(0, points).values())[0].T.reshape(
-                    points.shape[0], value_size, -1)[:, :, :15]
-        if element.name == "Bernardi-Raugel":
-            if example == "triangle,1":
-                for i in range(3):
-                    edofs[1][i] = edofs[1][i][:1]
-                return edofs, lambda points: list(e.tabulate(0, points).values())[0].T.reshape(
-                    points.shape[0], value_size, -1)[:, :, :9]
-            if example == "tetrahedron,1":
-                for i in range(4):
-                    edofs[2][i] = edofs[2][i][:1]
-                return edofs, lambda points: list(e.tabulate(0, points).values())[0].T.reshape(
-                    points.shape[0], value_size, -1)[:, :, :16]
-
-        return edofs, lambda points: list(e.tabulate(0, points).values())[0].T.reshape(
-            points.shape[0], value_size, -1)
+            reduced_dim = 15
+
+        if reduced_dim is not None:
+            for dim in range(len(edofs)):
+                for i in range(len(edofs[sd-1])):
+                    edofs[dim][i] = [dof for dof in edofs[dim][i] if dof < reduced_dim]
+
+            return edofs, lambda points: list(e.tabulate(0, points).values())[0][:reduced_dim].T
+
+        return edofs, lambda points: list(e.tabulate(0, points).values())[0].T
```

Comment:
Excellent work shortening this code, it was getting very messy

Reply:
I've just made it even shorter

Reply:
you also just pushed a fix to a bug I found when testing verification locally before I had time to post it!